worker/src/unstract/worker/worker.py

```diff
@@ -79,6 +80,9 @@ def process_log_message(
         if not log_dict:
             return None
         log_type = self.get_log_type(log_dict)
+        log_level = self.get_log_level(log_dict)
+        if log_type == LogType.LOG and log_level == LogLevel.ERROR:
+            raise ToolRunException(log_dict.get("log"))
```

Comment:
NIT: As discussed this might prevent us from using `ERROR` logs without exiting in the context of tools and SDK - we need to make sure we never catch and suppress an exception with ERROR logs and instead using WARNING instead. We could address this when necessary maybe

Reply:
Agreed @chandrasekharan-zipstack.
We might have to specify a proper identifier for stream_error_exit scenarios. Will take this up when such a requirement arises, should be quick.